src/Components/EditAdditionalRepositoriesTable/EditAdditionalRepositoriesTable.js

```diff
@@ -0,0 +1,429 @@
+        );
+        firstElement && firstElement.focus();
+      }
+    }, 0);
+    setIsAttributeMenuOpen(!isAttributeMenuOpen);
+  };
+  const attributeToggle = (
+    <MenuToggle
+      ref={attributeToggleRef}
+      onClick={onAttributeToggleClick}
+      isExpanded={isAttributeMenuOpen}
+      icon={<FilterIcon />}
+    >
+      {activeAttributeMenu}
+    </MenuToggle>
+  );
+
+  const attributeMenu = (
+    <Menu
+      ref={attributeMenuRef}
+      onSelect={(_ev, itemId) => {
+        setActiveAttributeMenu(itemId?.toString());
+        setIsAttributeMenuOpen(!isAttributeMenuOpen);
+      }}
+    >
+      <MenuContent>
+        <MenuList>
+          <MenuItem itemId="Name">Name</MenuItem>
+          <MenuItem itemId="Label">Label</MenuItem>
+        </MenuList>
+      </MenuContent>
+    </Menu>
+  );
+
+  const attributeDropdown = (
+    <div ref={attributeContainerRef}>
+      <Popper
+        trigger={attributeToggle}
+        popper={attributeMenu}
+        appendTo={attributeContainerRef.current || undefined}
+        isVisible={isAttributeMenuOpen}
+      />
+    </div>
+  );
+
+  const toolbarSearchInput = (
+    <SearchInput
+      placeholder={placeholderValue}
+      value={searchValue}
+      onChange={onSearchChange}
+      isDisabled={repositories?.length == 0}
+      onClear={() => onSearchChange('')}
+      style={{
+        width: '400px',
+      }}
+    />
+  );
+  /*Pagination*/
+
+  const getPage = (repo) => {
+    const first = (page - 1) * perPage;
+    const last = first + perPage;
+    return repo?.slice(first, last);
+  };
+
+  const paginatedRepos = getPage(repositories);
+  const pagination = (variant = PaginationVariant.bottom) => (
+    <Pagination
+      perPageComponent="button"
+      itemCount={repositories?.length}
+      perPage={perPage}
+      page={page}
+      onSetPage={onSetPage}
+      onPerPageSelect={onPerPageSelect}
+      isCompact
+      variant={variant}
+    />
+  );
+
+  /*Table sorting Filter by asc- desc*/
+
+  let sortedRepositories = repositories && filteredRepos && paginatedRepos;
+  if (activeSortIndex !== null) {
+    sortedRepositories = repositories.sort((a, b) => {
+      const aValue = getSortableRowValues(a)[activeSortIndex];
+      const bValue = getSortableRowValues(b)[activeSortIndex];
+      if (typeof aValue === 'number') {
+        if (activeSortDirection === 'asc') {
+          return aValue - bValue;
+        }
+        return bValue - aValue;
+      } else {
+        if (activeSortDirection === 'asc') {
+          return aValue.localeCompare(bValue);
+        }
+        return bValue.localeCompare(aValue);
+      }
+    });
+  }
+  const getSortParams = (columnIndex) => ({
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+      defaultDirection: 'asc',
+    },
+    onSort: (_event, index, direction) => {
+      setActiveSortIndex(index);
+      setActiveSortDirection(direction);
+    },
+    columnIndex,
```

Comment:
The sorting seems to be breaking the pagination. Again, I'd look how subscription inventory handles the sorting but there seems to be some sort of functions being used towrds the end that keeps pagination working

 const sortedProducts = sortProducts(data, activeSortIndex);
  const searchedProducts = filterDataBySearchTerm(sortedProducts, searchValue);
  const paginatedProducts = getPage(searchedProducts);
  
  These will need to be changed to fit the repos, but just pointing it out